packages/blade/src/components/Input/PasswordField/__tests__/PasswordField.web.test.tsx

```diff
@@ -0,0 +1,230 @@
+    const user = userEvent.setup();
+    const label = 'Enter password';
+    const userPassword = 'Divyanshu$123';
+    const onBlur = jest.fn();
+
+    renderWithTheme(
+      <PasswordField
+        label={label}
+        name="password"
+        defaultValue={userPassword}
+        // eslint-disable-next-line jsx-a11y/no-autofocus
+        autoFocus
+        onBlur={onBlur}
+      />,
+    );
+
+    // shifts user focus and therefore blurs the focussed input
+    await user.tab();
+    expect(onBlur).toHaveBeenCalledTimes(1);
+    expect(onBlur).toHaveBeenCalledWith({ name: 'password', value: userPassword });
+  });
+
+  it('should be focussable', async () => {
+    const user = userEvent.setup();
+    const label = 'Enter password';
+    const userPassword = 'Divyanshu$123';
+
+    const { getByLabelText } = renderWithTheme(
+      <PasswordField label={label} name="password" defaultValue={userPassword} />,
+    );
+
+    const input = getByLabelText(label);
+
+    expect(input).not.toHaveFocus();
+    await user.tab();
+    expect(input).toHaveFocus();
+  });
+
+  it('should set value as an uncontrolled input', async () => {
+    const user = userEvent.setup();
+    const label = 'Enter password';
+    const valueInitial = 'Divyanshu';
+    const valueFinal = 'Divyanshu$123';
+
+    const { getByLabelText } = renderWithTheme(
+      <PasswordField label={label} defaultValue={valueInitial} />,
+    );
+
+    const input = getByLabelText(label);
+    expect(input).toHaveValue(valueInitial);
+
+    await user.type(input, '$123');
+    expect(input).toHaveValue(valueFinal);
+  });
+
+  it('should set value as a controlled input', async () => {
+    const user = userEvent.setup();
+    const label = 'Enter password';
+    const valueInitial = 'Divyanshu';
+    const valueFinal = 'Divyanshu$123';
+
+    const ControlledInputExample = (): ReactElement => {
+      const [value, setValue] = useState<string | undefined>(valueInitial);
+
+      return (
+        <PasswordField label={label} value={value} onChange={({ value }) => setValue(value)} />
+      );
+    };
+
+    const { getByLabelText } = renderWithTheme(<ControlledInputExample />);
+
+    const input = getByLabelText(label);
+    expect(input).toHaveValue(valueInitial);
+
+    await user.type(input, '$123');
+    expect(input).toHaveValue(valueFinal);
+  });
+
+  it('should toggle password visibility on clicking reveal button', async () => {
+    const user = userEvent.setup();
+    const label = 'Enter password';
+    const userPassword = 'Divyanshu$123';
+
+    const { getByLabelText, getByRole } = renderWithTheme(
+      <PasswordField label={label} name="password" defaultValue={userPassword} />,
+    );
+
+    const input = getByLabelText(label);
+    const revealButton = getByRole('button');
+
+    expect(input).toHaveAttribute('type', 'password');
+    expect(revealButton).toHaveAccessibleName('Show password');
+
+    await user.click(revealButton);
```

Comment:
Should we also assert the focus? 

Like 

press.tab()
press.enter() 
revealButton.toHaveFocus()
press.enter()
revealButton.toHaveFocus()

Reply:
Good idea, added in https://github.com/razorpay/blade/pull/715/commits/20aa6d874a45a3494838b6b2dcabd860509c8e0b